src/main/java/com/javatechie/springboot/crud/example/controller/ProductController.java

```diff
@@ -1,25 +1,32 @@
 package com.javatechie.springboot.crud.example.controller;
 
+import com.javatechie.springboot.crud.example.dto.ProductDto;
 import com.javatechie.springboot.crud.example.entity.Product;
 import com.javatechie.springboot.crud.example.service.ProductService;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.web.bind.annotation.*;
 
+import java.util.ArrayList;
 import java.util.List;
 
+import static com.javatechie.springboot.crud.example.entity.ProductTransformer.mapProductDtoToProduct;
+
 @RestController
 public class ProductController {
 
  @Autowired
  private ProductService service;
 
  @PostMapping("/addProduct")
- public Product addProduct(@RequestBody Product product){
-    return service.saveProduct(product);
+ public Product addProduct(@RequestBody ProductDto productDto) throws Exception {
+    return service.saveProduct(mapProductDtoToProduct(productDto));
```

Comment:
Well done, sometimes its better to do not import the static method, instead you'll see something like  this:
    return service.saveProduct(**ProductTransformer.**mapProductDtoToProduct(productDto));